compiler/src/dotty/tools/dotc/transform/sjs/PrepJSInterop.scala

```diff
@@ -814,20 +822,41 @@ class PrepJSInterop extends MacroTransform with IdentityDenotTransformer { thisP
       super.transform(tree)
     }
 
+    /** Removes annotations from exported definitions (e.g. `export foo.bar`):
+     *  - `js.native`
+     *  - `js.annotation.*`
+     *  - `js.annotation.internal.*`
+     */
+    private def stripJSAnnotsOnExported(sym: Symbol)(using Context): Unit =
+      if !sym.is(Exported) then return
+
+      val JSNativeAnnot = jsdefn.JSNativeAnnot
+      val JSAnnotPackage = jsdefn.JSAnnotPackage
+      val JSAnnotInternalPackage = jsdefn.JSAnnotInternalPackage
+
+      extension (sym: Symbol) def isJSAnnot =
+        (sym eq JSNativeAnnot) || (sym.owner eq JSAnnotPackage) || (sym.owner eq JSAnnotInternalPackage)
+
+      val newAnnots = sym.annotations.filterConserve(!_.symbol.isJSAnnot)
+      if newAnnots ne sym.annotations then
+        sym.annotations = newAnnots
+    end stripJSAnnotsOnExported
+
     private def checkRHSCallsJSNative(tree: ValOrDefDef, longKindStr: String)(using Context): Unit = {
-      // Check that the rhs is exactly `= js.native`
-      tree.rhs match {
-        case sel: Select if sel.symbol == jsdefn.JSPackage_native =>
-          // ok
-        case _ =>
-          val pos = if (tree.rhs != EmptyTree) tree.rhs.srcPos else tree.srcPos
-          report.error(s"$longKindStr may only call js.native.", pos)
-      }
+      if !tree.symbol.is(Exported) then
```

Comment:
Is this change needed? The other rules forbid an `export` in places where `checkRHSCallsJSNative` would be called.

Unless this prevents hard-to-understand cascading errors? If that's the reason, please say so in a comment.

Reply:
This is required because this code path will be reached whenever there is a definition in a JS Native type - so it is to prevent getting an extra error